handler/ndnp.py

```diff
@@ -137,26 +139,38 @@ class Batch():
 
     '''iterator class representing the set of resources to be loaded'''
 
-    def __init__(self, args):
+    def __init__(self, config):
         self.logger = logging.getLogger(
             __name__ + '.' + self.__class__.__name__
-            )
-        tree = ET.parse(args.path)
-        root = tree.getroot()
-        m = XPATHMAP
 
-        dback = Collection()
-        dback.title = "The Diamondback Newspaper Collection"
-        dback.graph.add(
-            (dback.uri, dcterms.title, rdflib.Literal(dback.title))
             )
+        self.batchfile = config.get('LOCAL_PATH')
+        self.collection = Collection()
+        self.collection.uri = rdflib.URIRef(config.get('COLLECTION'))
```

Comment:
If I create a batch.yml with an empty `COLLECTION` entry, this line throws an exception, since the `config.get('COLLECTION')` returns `None`. If a collection is required, I think it would be better to check for the existence of the `COLLECTION` key and abort the load if it is not present. If a collection isn't required, then we need to handle that case.